### PR TITLE
8305716: Enhancements for printing age tables

### PR DESCRIPTION
**Pull request by earthling-amzn**
In order to reduce the number of changes for the upcoming PR for the generational mode for Shenandoah we are opening separate PRs for the changes in common code. Shenandoah will be using this `AgeTable` class. There are two changes here:
* `AgeTables` created with global=false will no longer SEGV when printing
* A method was added to allow the `AgeTable` to print to an existing output stream

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8305716](https://bugs.openjdk.org/browse/JDK-8305716): Enhancements for printing age tables

### Reviewers
 * [Kelvin Nilsen](https://openjdk.org/census#kdnilsen) (@kdnilsen - no project role) ⚠️ Review applies to [0c1fcefa](https://git.openjdk.org/jdk/pull/13377/files/0c1fcefaa6d7e5bf2371254dc10ae66cb0504585)
 * [Y. Srinivas Ramakrishna](https://openjdk.org/census#ysr) (@ysramakrishna - **Reviewer**) ⚠️ Review applies to [0c1fcefa](https://git.openjdk.org/jdk/pull/13377/files/0c1fcefaa6d7e5bf2371254dc10ae66cb0504585)
 * [Thomas Schatzl](https://openjdk.org/census#tschatzl) (@tschatzl - **Reviewer**) ⚠️ Review applies to [0c1fcefa](https://git.openjdk.org/jdk/pull/13377/files/0c1fcefaa6d7e5bf2371254dc10ae66cb0504585)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13377/head:pull/13377` \
`$ git checkout pull/13377`

Update a local copy of the PR: \
`$ git checkout pull/13377` \
`$ git pull https://git.openjdk.org/jdk.git pull/13377/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13377`

View PR using the GUI difftool: \
`$ git pr show -t 13377`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13377.diff">https://git.openjdk.org/jdk/pull/13377.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13377#issuecomment-1499456633)

### CODE REVIEW

**Review by kdnilsen — APPROVED** [review]
Thanks for improved generality.

**Review by ysramakrishna — APPROVED** [review]
LGTM ... 🚢

File: src/hotspot/share/gc/shared/ageTable.cpp [hunk 1]
```diff
@@ -105,25 +106,30 @@ uint AgeTable::compute_tenuring_threshold(size_t desired_survivor_size) {
...
+  LogTarget(Trace, gc, age) lt;
+  if (lt.is_enabled() || _use_perf_data || AgeTableTracer::is_tenuring_distribution_event_enabled()) {
+    LogStream st(lt);
+    print_on(&st, tenuring_threshold);
   }
 }
 
+void AgeTable::print_on(outputStream* st, uint tenuring_threshold) {
+  st->print_cr("Age table with threshold %u (max threshold " UINTX_FORMAT ")",
+           tenuring_threshold, MaxTenuringThreshold);
```
Comment (tschatzl) [inline src/hotspot/share/gc/shared/ageTable.cpp:118]:
```suggestion
  st->print_cr("Age table with threshold %u (max threshold " UINTX_FORMAT ")",
               tenuring_threshold, MaxTenuringThreshold);
```
arguments should be aligned after the bracket above

**Review by tschatzl — APPROVED** [review]
Lgtm.
Please fix that indentation issue before pushing.

**Pull request closed without merging**